packages/at_client/lib/src/service/sync_service_impl.dart

```diff
@@ -572,10 +572,11 @@ class SyncServiceImpl implements SyncService, AtSignChangeListener {
     final key = serverCommitEntry['atKey'];
     // publickey.<atsign>@<currentatsign> is used to store the public key of
     // other atsign. The value is not encrypted.
-    // The keys starting with publickey. and shared_key. are the reserved keys
+    // The keys starting with publickey. and keys that contain shared_key
+    // (@someone:shared_key@me, shared_key.someone@me) are the reserved keys
     // and do not require actions. Hence skipping from checking conflict resolution.
     if (key.startsWith('publickey.') ||
-        key.startsWith('shared_key.') ||
+        key.contains('shared_key') ||
```

Comment:
Let's be a bit more precise - we should match on the key looking like either `@alice:shared_key@bob` or `shared_key.alice@bob`

Reply:
@gkc: How about rewriting the conditions as `key.startsWith(RegExp('shared_key.+@.+|@.+shared_key@.+'))`